python/sklearn.Dockerfile

```diff
@@ -1,7 +1,14 @@
 FROM python:3.7-slim
 
+# gsutil required packages
+RUN apt-get update && apt-get install -y curl lsb-release gnupg
+RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+RUN apt-get update && apt-get install -y google-cloud-sdk
```

Comment:
There is no python version for google cloud sdk - available with pip install?

Reply:
There is, I used it when the user provide the `GOOGLE_APPLICATION_CREDENTIALS` for their own bucket. For public accessible data, I only can pull it using either gsutil or rest API. You can visit the details at https://cloud.google.com/storage/docs/access-public-data.

If anyone found a solution to access gcs public data using the python SDK, I would happy to implement that.

Reply:
Also, the pip install version of gsutil only works on python 2.7 and our image is using python 3.7

Reply:
https://pypi.org/project/google-cloud-storage/

Reply:
Just to be clear over here, the google cloud python client is available on python 3.7. However, we need the gsutil command line client to access public data, and the pip install for gsutil is python 2.7 only https://cloud.google.com/storage/docs/gsutil_install#alt-install.

Therefore, I went with installing the whole google cloud command line SDK which won't depends on the local machine python version.

Reply:
Definitely not ideal, but I think what you've done is probably right. I want to double check there isn't something better; I'll poke around.

Reply:
@ellis-bigelow is that the normal user experience on GCS? Two different SDKs based on scenarios?

Reply:
I think we might be able to use https://googleapis.github.io/google-cloud-python/latest/storage/client.html#google.cloud.storage.client.Client.create_anonymous_client

Reply:
Let`s jus stick to gsutil then. Less error prone

Reply:
I take a look at the google.cloud.storage source code, looks like with anonymous client and `.bucket()` will able to get the object without credentials. I think the google cloud docs is somewhat misleading the users to use `.get_bucket()` which required access credentials. 
https://cloud.google.com/storage/docs/downloading-objects#storage-download-object-code_sample
Also the access-public-data docs doesn't show any client SDK example except gsutil.

Let me investigate a bit more and see can I do it in pure python SDK. If not, then we can stick back with gsutil.

Reply:
@ellis-bigelow So I did find a way to access public data with the Python client using the anonymous client. I reimplemented the gcs function with pure gcs python SDK now.